CI failing on staticcheck version

Failing on HEAD for all `tip` configurations (linux and macos), when trying to invoke `staticcheck`:

https://github.com/google/pprof/runs/4063340955
```
++ staticcheck github.com/google/pprof github.com/google/pprof/driver github.com/google/pprof/fuzz github.com/google/pprof/internal/binutils github.com/google/pprof/internal/driver github.com/google/pprof/internal/elfexec github.com/google/pprof/internal/graph github.com/google/pprof/internal/measurement github.com/google/pprof/internal/plugin github.com/google/pprof/internal/proftest github.com/google/pprof/internal/report github.com/google/pprof/internal/symbolizer github.com/google/pprof/internal/symbolz github.com/google/pprof/internal/transport github.com/google/pprof/profile github.com/google/pprof/third_party/d3 github.com/google/pprof/third_party/d3flamegraph github.com/google/pprof/third_party/svgpan
-: cannot import "internal/cpu" (unknown iexport format version 2), export data is newer version - update tool (compile)
-: cannot import "internal/goarch" (unknown iexport format version 2), export data is newer version - update tool (compile)
Error: Process completed with exit code 1.
```

It seems like a simple version issue with `staticcheck` -- probably because we don't use `actions/setup-go@v2` in the `tip` configuration [here](https://github.com/google/pprof/blob/master/.github/workflows/ci.yaml#L112) and instead fetch lint tools [manually](https://github.com/google/pprof/blob/master/.github/workflows/ci.yaml#L138).

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -50,13 +50,12 @@ jobs:
       - name: Update Go version manually
         if: matrix.go == 'tip'
         working-directory: ${{ github.workspace }}
-        env: 
-          RUN_STATICCHECK: false
         run: |
           git clone https://go.googlesource.com/go $HOME/gotip
           cd $HOME/gotip/src
           ./make.bash
           echo "GOROOT=$HOME/gotip" >> $GITHUB_ENV
+          echo "RUN_STATICCHECK=false" >> $GITHUB_ENV
           echo "$HOME/gotip/bin:$PATH" >> $GITHUB_PATH
 
       - name: Checkout the repo
@@ -118,14 +117,13 @@ jobs:
 
       - name: Update Go version manually
         if: matrix.go == 'tip'
-        env: 
-          RUN_STATICCHECK: false
         working-directory: ${{ github.workspace }}
         run: |
           git clone https://go.googlesource.com/go $HOME/gotip
           cd $HOME/gotip/src
           ./make.bash
           echo "GOROOT=$HOME/gotip" >> $GITHUB_ENV
+          echo "RUN_STATICCHECK=false" >> $GITHUB_ENV
           echo "$HOME/gotip/bin" >> $GITHUB_PATH
 
       - name: Checkout the repo

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -50,6 +50,8 @@ jobs:
       - name: Update Go version manually
         if: matrix.go == 'tip'
         working-directory: ${{ github.workspace }}
+        env: 
+          RUN_STATICCHECK: false
         run: |
           git clone https://go.googlesource.com/go $HOME/gotip
           cd $HOME/gotip/src
@@ -116,6 +118,8 @@ jobs:
 
       - name: Update Go version manually
         if: matrix.go == 'tip'
+        env: 
+          RUN_STATICCHECK: false
         working-directory: ${{ github.workspace }}
         run: |
           git clone https://go.googlesource.com/go $HOME/gotip

--- a/test.sh
+++ b/test.sh
@@ -23,7 +23,9 @@ echo "mode: $MODE" > coverage.txt
 # All packages.
 PKG=$(go list ./...)
 
-staticcheck $PKG
+if [ "$RUN_STATICCHECK" != "false" ]; then
+  staticcheck $PKG
+fi
 
 # Packages that have any tests.
 PKG=$(go list -f '{{if .TestGoFiles}} {{.ImportPath}} {{end}}' ./...)

